build-system/tasks/process-github-issues.js

```diff
@@ -0,0 +1,274 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+var BBPromise = require('bluebird');
+var argv = require('minimist')(process.argv.slice(2));
+var assert = require('assert');
+var child_process = require('child_process');
+var config = require('../config');
+var extend = require('util')._extend;
+var fs = require('fs-extra');
+var git = require('gulp-git');
+var gulp = require('gulp-help')(require('gulp'));
+var request = BBPromise.promisify(require('request'));
+var util = require('gulp-util');
+
+var GITHUB_ACCESS_TOKEN = process.env.GITHUB_ACCESS_TOKEN;
+var exec = BBPromise.promisify(child_process.exec);
+var gitExec = BBPromise.promisify(git.exec);
+
+var isDryrun = argv.dryrun;
+var verbose = (argv.verbose || argv.v);
+
+const issuesOpt = {
+  url: 'https://api.github.com/repos/ampproject/amphtml/issues',
+  headers: {
+    'User-Agent': 'amp-changelog-gulp-task',
+    'Accept': 'application/vnd.github.v3+json'
+  },
+};
+
+const optionsMilestone = {
+  url: 'https://api.github.com/repos/ampproject/amphtml/milestones',
+  headers: {
+    'User-Agent': 'amp-changelog-gulp-task',
+    'Accept': 'application/vnd.github.v3+json'
+  },
+};
+
+if (GITHUB_ACCESS_TOKEN) {
+  issuesOpt.qs = {
+    access_token: GITHUB_ACCESS_TOKEN
+  }
+  optionsMilestone.qs = {
+    access_token: GITHUB_ACCESS_TOKEN
+  }
+}
+
+
+function processIssues() {
+  if (!GITHUB_ACCESS_TOKEN) {
+    util.log(util.colors.red('You have not set the ' +
+        'GITHUB_ACCESS_TOKEN env var.'));
+    util.log(util.colors.green('See https://help.github.com/articles/' +
+        'creating-an-access-token-for-command-line-use/ ' +
+        'for instructions on how to create a github access token. We only ' +
+        'need `public_repo` scope.'));
+    return;
+  }
+  return getGitMetadata();
+}
+
+function getGitMetadata() {
+  return getIssues().then(function() {
+    util.log(util.colors.blue('automation applied'));
+  })
+}
+
+/**
+ * Function goes through all the gitHub issues,
+ * gets all the Labels we are interested in,
+ * depending if missing milestone or label,
+ * tasks applied as per design go/ampgithubautomation
+ */
+function getIssues(){
```

Comment:
This function appears to be doing much more than a getter. Suggest using a more descriptive name like `updateGithubIssues`.